packages/api/src/task/task.ts

```diff
@@ -84,21 +88,65 @@ export default class TaskScheduler {
             status: "ready",
           });
 
-          await db.task.update(task.id, {
-            status: {
-              phase: "completed",
-              updatedAt: Date.now(),
-            },
-            output: event.output,
-          });
           return true;
         }
       }
-      console.log(`task type unknown: ${event.task.type}`);
+
+      if (allowedTasks.includes(event.task.type)) {
+        await db.task.update(task.id, {
+          status: {
+            phase: "completed",
+            updatedAt: Date.now(),
+          },
+          output: event.output,
+        });
+        return true;
+      } else {
+        console.log(`task type unknown: ${event.task.type}`);
+      }
     } else {
       console.log(`task event process error: task ${event.task.id} not found`);
       return true;
     }
     return false;
   }
+
+  async scheduleTask(
+    asset: Asset,
+    type: "import" | "export" | "transcode",
+    params: object
+  ) {
+    let newTask: WithID<Task> = {
+      id: uuid(),
+      name: `asset-upload-${asset.name}-${asset.createdAt}`,
```

Comment:
Name is inaccurate for the other task types. Either receive it as an argument to the function here or try to use the task type to build the name instead? 

From another perspective, feels like this name is unnecessary, perhaps something we'll just want to compose on the frontend but doesn't have any meaning/necessity in the backend. WDYT of just removing it?